### PR TITLE
add network cancelation point and fix timeout handle logic in sys_arch.c

### PR DESCRIPTION
**Pull request by lssgood**
1. Add the network cancelation point in sys_arch.c
2. Fix an errno setting method in sem_tickwait.c
3. Fix timeout handle logic in sys_arch_sem_wait()

### CODE REVIEW

File: os/net/lwip/sys/arch/sys_arch.c [hunk 1]
```diff
@@ -128,19 +129,25 @@ void sys_mbox_post(sys_mbox_t *mbox, void *msg)
...
+		return;
+	}
 	LWIP_DEBUGF(SYS_DEBUG, ("mbox %p msg %p\n", (void *)mbox, (void *)msg));
 	/* Wait while the queue is full */
 	tmp = (mbox->rear + 1) % mbox->queue_size;
 	if (tmp == mbox->front) {
 		LWIP_DEBUGF(SYS_DEBUG, ("Queue Full, Wait until gets free\n"));
 	}
 	while (tmp == mbox->front) {
+		if (status == SYS_ARCH_CANCELED) {
```
Comment (sunghan-chang) [inline os/net/lwip/sys/arch/sys_arch.c]: I think you had better put this at end of while, not beginnig of while.
When status is cancelled case, unncessary operation will be occured. (checking tmp == mbox->front)

File: os/net/lwip/sys/arch/sys_arch.c [hunk 2]
```diff
@@ -237,19 +244,28 @@ err_t sys_mbox_trypost(sys_mbox_t *mbox, void *msg)
...
 	   stuff here. */
-	sys_arch_sem_wait(&(mbox->mutex), 0);
+	status = sys_arch_sem_wait(&(mbox->mutex), 0);
+	if (status == SYS_ARCH_CANCELED) {
+		return SYS_ARCH_CANCELED;
+	}
 
 	/* wait while the queue is empty */
 	while (mbox->front == mbox->rear) {
+		if (status == SYS_ARCH_CANCELED) {
```
Comment (sunghan-chang) [inline os/net/lwip/sys/arch/sys_arch.c]: same as previous

File: os/net/socket/bsd_socket_api.c [hunk 3]
```diff
@@ -54,96 +54,138 @@
...
+#include <tinyara/cancelpt.h>
 
 #ifdef CONFIG_NET
 
 #include <sys/socket.h>
 
 int bind(int s, const struct sockaddr *name, socklen_t namelen)
 {
-	return lwip_bind(s, name, namelen);
+	int result = lwip_bind(s, name, namelen);
```
Comment (sunghan-chang) [inline os/net/socket/bsd_socket_api.c]: why do you need to define result for same operation?

File: os/net/socket/bsd_socket_api.c [hunk 4]
```diff
@@ -54,96 +54,138 @@
...
+	(void)enter_cancellation_point();
+	int result = lwip_accept(s, addr, addrlen);
+	leave_cancellation_point();
+	return result;
 }
 
 int shutdown(int s, int how)
 {
-	return lwip_shutdown(s, how);
+	int result = lwip_shutdown(s, how);
```
Comment (sunghan-chang) [inline os/net/socket/bsd_socket_api.c]: same as previous

File: os/net/socket/bsd_socket_api.c [hunk 5]
```diff
@@ -54,96 +54,138 @@
...
+	(void)enter_cancellation_point();
+	int result = lwip_connect(s, name, namelen);
+	leave_cancellation_point();
+	return result;
 }
 
 int getsockname(int s, struct sockaddr *name, socklen_t *namelen)
 {
-	return lwip_getsockname(s, name, namelen);
+	int result = lwip_getsockname(s, name, namelen);
```
Comment (sunghan-chang) [inline os/net/socket/bsd_socket_api.c]: same

File: os/net/lwip/sys/arch/sys_arch.c (hunk 1 above)
Comment (HONGCHAEHEE) [inline os/net/lwip/sys/arch/sys_arch.c]: I think so. I have updated to put this part at the end of while.

File: os/net/socket/bsd_socket_api.c (hunk 3 above)
Comment (HONGCHAEHEE) [inline os/net/socket/bsd_socket_api.c]: removed the unnecessary definition

**Pull request merged**